### PR TITLE
Add SyntaxErrors in GlobalDeclarationInstantiation

### PR DESCRIPTION
**Pull request by raskad**
This Pull Request  changes the following:

- Add SyntaxErrors in GlobalDeclarationInstantiation.
- Remove the unused `Early` phase in `boa_tester`.
- Move the compilation from the `parse` to the `runtime` phase in `boa_tester`.

### CODE REVIEW

**Review by HalidOdat — APPROVED** [review]
Looks great to me! :)

File: boa_engine/src/bytecompiler/declarations.rs [hunk 1]
```diff
@@ -26,16 +26,44 @@ impl ByteCompiler<'_, '_> {
...
+                    .into());
+            }
+
+            // c. Let hasRestrictedGlobal be ? env.HasRestrictedGlobalProperty(name).
+            let has_restricted_global = self.context.has_restricted_global_property(name)?;
+
+            // d. If hasRestrictedGlobal is true, throw a SyntaxError exception.
+            if has_restricted_global {
+                return Err(JsNativeError::syntax()
+                    .with_message("restricted global property")
```
Comment (jedel1043) [inline boa_engine/src/bytecompiler/declarations.rs:53]:
```suggestion
                    .with_message("cannot redefine non-configurable global property")
```

**Review by jedel1043 — COMMENTED** [review]
Everything looking good! Just had a small nitpick with an error message

File: boa_engine/src/bytecompiler/declarations.rs (hunk 1 above)
Comment (raskad) [reply boa_engine/src/bytecompiler/declarations.rs]: Fixed

**Review by jedel1043 — APPROVED** [review]

**Pull request merged**